catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java

```diff
@@ -532,114 +378,251 @@ private boolean isEmail(Destination dest) {
                   contents,
                   metacards);
             }
-            break;
-          case TARZIP:
-            {
-              try (TemporaryFileBackedOutputStream tarFos =
-                      new TemporaryFileBackedOutputStream(MAX_MEMORY_SIZE);
-                  TarOutputStream tarOut = new TarOutputStream(tarFos)) {
-                getTar(tarOut, file);
-                try (TemporaryFileBackedOutputStream zipFos =
-                        new TemporaryFileBackedOutputStream(MAX_MEMORY_SIZE);
-                    ZipOutputStream zipOut = new ZipOutputStream(zipFos)) {
-                  getZip(zipOut, tarFos.asByteSource().openStream(), filename + ".tar");
-                  ByteSource contents = zipFos.asByteSource();
-
-                  writeFile(
-                      destinationSink,
-                      packagingSpecFormatType,
-                      filename,
-                      sentFiles,
-                      contents,
-                      metacards);
-                }
+          }
+        }
+        break;
+      case FILESGZIP:
+        try (TemporaryFileBackedOutputStream fos =
+                new TemporaryFileBackedOutputStream(MAX_MEMORY_SIZE);
+            GZIPOutputStream zipOut = new GZIPOutputStream(fos);
+            InputStream fileInputStream = file.getInputStream()) {
+          getGzip(zipOut, fileInputStream);
+          ByteSource contents = fos.asByteSource();
+          writeFile(
+              destinationSink, packagingSpecFormatType, filename, sentFiles, contents, metacards);
+        }
+        break;
+      case TARCOMPRESS:
+        {
+          writeTarFile(
+              destinationSink, packagingSpecFormatType, filename, sentFiles, file, metacards);
+        }
+        break;
+      default:
+        break;
```

Comment:
✏️ Maybe we could do in a different PR, but should we be logging when we hit this default something along the lines of "Unknown packaging type, skipping".

Reply:
I can add